### PR TITLE
Added client side certificate and key pair.

### PR DESCRIPTION
**Pull request by tillt**
- adds the ability to provide client side SSL certificate/key-pair for the TLS configuration
- adds configuration parsing for `CertFile`
- adds configuration parsing for `KeyFile`

### CODE REVIEW

File: records/config.go [hunk 1]
```diff
@@ -165,6 +173,14 @@ func SetConfig(cjson string) Config {
 		c.caPool = pool
 	}
 
+	if c.CertFile != "" {
```
Comment (jdef) [inline records/config.go]: it's probably an error if one of these new config params is set, but not the other. any value in having a smarter check here for sanity?

Comment (tillt) [inline records/config.go]: Jeez, this ```records/config.go:134::warning: cyclomatic complexity 16 of function SetConfig() is high (> 12) (gocyclo)``` is getting me crazy - guess I should leave this stuff to the pro's.

**Comment by jdef** [comment]
lgtm. PTAL @sargun

**Comment by sargun** [comment]
LGTM. I'll merge once you rebase.

**Comment by tillt** [comment]
@sargun is there more I need to do?

**Pull request merged**